### PR TITLE
[Doc-Improvement] Add CDC-Ingestion section

### PR DESCRIPTION
**Pull request by lklhdu**
## Why are the changes needed?

*CDC ingestion is a common application scenario that requires a documented description to guide practice in this scenario.*

## Brief change log

  - *Add the CDC ingestion documentation in the user-guides chapter*

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

File: docs/user-guides/cdc-ingestion.md [hunk 1]
```diff
@@ -8,4 +8,500 @@ menu:
...
+);
+
+insert into `arctic_catalog`.`db`.`test_tb` select * from products;
+```
+
+## Multiple tables into the lake
+### Native Iceberg format
+The following example will show how to write CDC data from multiple MySQL tables into the corresponding Native-Iceberg table.
+```java
+import com.ververica.cdc.connectors.mysql.source.MySqlSource;
```
Comment (YesOrNo828) [inline docs/user-guides/cdc-ingestion.md:88]: I think we can remove the import, and add maven or gradle dependency detail at the head of demos

File: docs/user-guides/cdc-ingestion.md [hunk 2]
```diff
@@ -8,4 +8,500 @@ menu:
...
+
+    public RowDataVoidProcessFunction(final Map<String, RowRowConverter> converterMap) {
+      this.converters = converterMap;
+    }
+
+    @Override
+    public void processElement(final RowData rowData,
+                               final ProcessFunction<RowData, Void>.Context ctx, final Collector<Void> out)
+      throws Exception {
+      final String key = rowData.getString(rowData.getArity() - 2).toString();
```
Comment (YesOrNo828) [inline docs/user-guides/cdc-ingestion.md:439]: why minus 2?

File: docs/user-guides/cdc-ingestion.md [hunk 3]
```diff
@@ -8,4 +8,500 @@ menu:
...
+      .fromSource(mySqlSource, WatermarkStrategy.noWatermarks(), "MySQL Source").setParallelism(4)
+      .process(new RowDataVoidProcessFunction(pathAndTable.stream()
+        .collect(toMap(e -> e.f0.toString(),
+          e -> RowRowConverter.create(e.f1.getResolvedSchema().toPhysicalRowDataType())))))
+      .name("split stream");
+
+    // create Arctic sink and insert into cdc data
+    InternalCatalogBuilder catalogBuilder = InternalCatalogBuilder.builder().metastoreUrl(
+      "thrift://<ip>:<port>/<catalog_name_in_metastore>");
+    Map<String, TableSchema> tableSchemas = new HashMap<>();
```
Comment (YesOrNo828) [inline docs/user-guides/cdc-ingestion.md:379]: How about renaming to sinkTableSchemas?

File: docs/user-guides/cdc-ingestion.md [hunk 4]
```diff
@@ -8,4 +8,500 @@ menu:
...
+
+    public RowDataVoidProcessFunction(final Map<String, RowRowConverter> converterMap) {
+      this.converters = converterMap;
+    }
+
+    @Override
+    public void processElement(final RowData rowData,
+                               final ProcessFunction<RowData, Void>.Context ctx, final Collector<Void> out)
+      throws Exception {
+      final String key = rowData.getString(rowData.getArity() - 2).toString();
```
Comment (YesOrNo828) [inline docs/user-guides/cdc-ingestion.md:231]: why minus 2?

**Review by YesOrNo828 — COMMENTED** [review]
Could the demos of multiple tables ingested into the data lake be more simplified?
 e.g. initSourceTables method.

File: docs/user-guides/cdc-ingestion.md [hunk 5]
```diff
@@ -8,4 +8,435 @@ menu:
         parent: User Guides
         weight: 400
 ---
-# CDC Ingestion
\ No newline at end of file
+# CDC Ingestion
+CDC stands for Change Data Capture, which is a broad concept, as long as it can capture the change data, it can be called CDC. [Flink CDC](https://github.com/ververica/flink-cdc-connectors) is a Log message-based data capture tool, all the inventory and incremental data can be captured. Taking MySQL as an example, it can easily capture Binlog data through Debezium and process the calculations in real time to send them to the Arctic data lake. The Arctic data lake can then be queried by other engines.
```
Comment (zhoujinsong) [inline docs/user-guides/cdc-ingestion.md:12]: Arctic's positioning is Lakehouse management service, `Arctic data lake` is an inaccurate statement.I suggest changing it directly to "data lake".

File: docs/user-guides/cdc-ingestion.md [hunk 6]
```diff
@@ -8,4 +8,435 @@ menu:
...
+
+This section will show how to practice one table into the lake and multiple tables into the lake for both [Native-Iceberg](../formats/iceberg.md) and [Mixed-Iceberg](../formats/mixed-iceberg.md) format.
```
Comment (zhoujinsong) [inline docs/user-guides/cdc-ingestion.md:14]:
```suggestion
This section will show how to ingest one table or multiple tables into the data lake for both [Iceberg](../formats/iceberg.md) format and [Mixed-Iceberg](../formats/mixed-iceberg.md) format.
```

File: docs/user-guides/cdc-ingestion.md [hunk 7]
```diff
@@ -8,4 +8,435 @@ menu:
...
+## One table into the lake
```
Comment (zhoujinsong) [inline docs/user-guides/cdc-ingestion.md:15]:
```suggestion
## Ingest into one table
```

File: docs/user-guides/cdc-ingestion.md [hunk 8]
```diff
@@ -8,4 +8,435 @@ menu:
...
+### Native Iceberg format
```
Comment (zhoujinsong) [inline docs/user-guides/cdc-ingestion.md:16]:
```suggestion
### Iceberg format
```

File: docs/user-guides/cdc-ingestion.md [hunk 9]
```diff
@@ -8,4 +8,435 @@ menu:
...
+The following example will show how MySQL CDC data is written to a Native-Iceberg table.
```
Comment (zhoujinsong) [inline docs/user-guides/cdc-ingestion.md:17]:
```suggestion
The following example will show how MySQL CDC data is written to a Iceberg table.
```

File: docs/user-guides/cdc-ingestion.md [hunk 10]
```diff
@@ -8,4 +8,435 @@ menu:
...
+
+INSERT INTO `iceberg_hadoop_catalog`.`default`.`sample` SELECT * FROM products;
+```
+
+### Mixed Iceberg format
+The following example will show how MySQL CDC data is written to a Mixed-Iceberg table.
+
+**Requirements**
+
+Please add [Flink Connector MySQL CDC](https://repo1.maven.org/maven2/com/ververica/flink-connector-mysql-cdc/2.3.0/flink-connector-mysql-cdc-2.3.0.jar) and [Arctic](https://repo1.maven.org/maven2/com/netease/arctic/arctic-flink-runtime-1.14/0.4.1/arctic-flink-runtime-1.14-0.4.1.jar) Jars to the lib directory of the Flink engine package.
```
Comment (zhoujinsong) [inline docs/user-guides/cdc-ingestion.md:61]:
```suggestion
Please add [Flink Connector MySQL CDC](https://repo1.maven.org/maven2/com/ververica/flink-connector-mysql-cdc/2.3.0/flink-connector-mysql-cdc-2.3.0.jar) and [Arctic](../../download) Jars to the lib directory of the Flink engine package.
```

File: docs/user-guides/cdc-ingestion.md [hunk 11]
```diff
@@ -8,4 +8,435 @@ menu:
...
+    id INT,
+    name STRING,
+    description STRING,
+    PRIMARY KEY (id) NOT ENFORCED
+);
+
+INSERT INTO `arctic_catalog`.`db`.`test_tb` SELECT * FROM products;
+```
+
+## Multiple tables into the lake
```
Comment (zhoujinsong) [inline docs/user-guides/cdc-ingestion.md:94]:
```suggestion
## Ingest Into multiple tables
```

File: docs/user-guides/cdc-ingestion.md [hunk 12]
```diff
@@ -8,4 +8,435 @@ menu:
...
+### Native Iceberg format
```
Comment (zhoujinsong) [inline docs/user-guides/cdc-ingestion.md:95]:
```suggestion
### Iceberg format
```

File: docs/user-guides/cdc-ingestion.md [hunk 13]
```diff
@@ -8,4 +8,435 @@ menu:
...
+The following example will show how to write CDC data from multiple MySQL tables into the corresponding Native-Iceberg table.
+
+**Requirements**
+
+Please add [Flink Connector MySQL CDC](https://mvnrepository.com/artifact/com.ververica/flink-connector-mysql-cdc/2.3.0) and [Iceberg](https://mvnrepository.com/artifact/org.apache.iceberg/iceberg-flink-1.14/1.1.0) dependencies to your Maven project's pom.xml file.
+
+```java
```
Comment (zhoujinsong) [inline docs/user-guides/cdc-ingestion.md:102]: add import statements at first

File: docs/user-guides/cdc-ingestion.md [hunk 14]
```diff
@@ -8,4 +8,435 @@ menu:
...
+      deserializationSchemaMap.get(db + "." + table).deserialize(record, out);
+    }
+
+    @Override
+    public TypeInformation<RowData> getProducedType() {
+      return TypeInformation.of(RowData.class);
+    }
+  }
+
+  static class RowDataVoidProcessFunction extends ProcessFunction<RowData, Void> {
```
Comment (zhoujinsong) [inline docs/user-guides/cdc-ingestion.md:187]: Give a more specific name to the class.

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Pull request merged**